<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
    <!-- Arguments -->
    <arg name="log_level" default="INFO" />
    <arg name="rpi_model" default="4"/>

    <group ns="niryo_robot_rpi">
        <param name="log_level" type="string" value="$(arg log_level)" />
        <rosparam file="$(find niryo_robot_rpi)/config/default.yaml"/>
        <param name="rpi_model" type="string" value="$(arg rpi_model)"/>

        <param name="publish_io_state_frequency" type="double" value="10.0"/>

        <param name="ros_log_size_threshold" type="int" value="1024"/> <!-- 1024 MB -->
        <param name="should_purge_ros_log_on_startup_file" type="string"
               value="/home/niryo/niryo_robot_saved_files/purge_ros_log_at_startup.txt" unless="$(eval rpi_model == 'simulation')"/>

        <!-- don't change that unless you manually change ros log location -->
        <param name="ros_log_location" type="string" value="$(env HOME)/.ros/log" />
    </group>
    
</launch>